---
layout: default
---
<div class="home">
  <div class="materials-wrap">
    <p style="text-align: center">Additional course websites: <a href="https://stellar.mit.edu/S/course/6/sp19/6.874/">MIT Stellar</a> | <a href="https://stellar.mit.edu/S/course/6/sp19/6.874/gradebook3/">Learning Modules</a> (where you download problem set assignments and upload your solutions) | <a href="https://piazza.com/mit/spring2019/6802j6874j20390j20490jhst506j">Piazza</a> (discussion forum)</p>
    <h2 class="module-header">Course description</h2>
    <p>Welcome to a new approach to computational problems in the life sciences. This subject is not an encyclopedic summary of contemporary methods in systems biology and genomics. Rather, we will explore both conventional and deep learning approaches to key problems in the life sciences, comparing and contrasting their power and limits.</p>

    <p>Our aim is to enable you to evaluate a wide variety of solutions to key problems you will face in this rapidly developing field, and enable you to execute on new enabling solutions that can have large impact.</p>
    
    <p>As part of the subject you will become an expert in using modern cloud resources to implement your solutions to challenging problems, first in problem sets that span a carefully chosen set of tasks, and then in an independent project. You will be programming using Python 3 and TensorFlow 1.12 in Jupyter Notebooks on the Google Cloud, a nod to the importance of carefully documenting your work so it can be precisely reproduced by others.</p>

	<h2 class="module-header">Syllabus</h2>
    <table class="table">
        <tr class="active">
            <th>&nbsp;</th><th>Date</th><th>Description</th><th>Course materials</th><th>Reference</th>
        </tr>
        <tr>
            <td>Lecture 1</td>
            <td>Feb 5</td>
            <td>Machine learning in the computational life sciences</td>
            <td>
                <ul>
                    <li>Read <a href="http://www.deeplearningbook.org/">Goodfellow</a> Chapter 1</li>
                </ul>
            </td>
            <td>
                <ul>
                    <li><a href="https://arxiv.org/pdf/1603.06430.pdf">DL in Bioinformatics</a></li>
                    <li><a href="http://msb.embopress.org/content/msb/12/7/878.full.pdf">DL for computational biology</a></li>
                    <li><a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1000809">The Roots of Bioinformatics</a></li>
                    <li><a href="http://ieeexplore.ieee.org/stamp/stamp.jsp?reload=true&arnumber=7347331">ML in Genomic Medicine</a></li>
                    <li><a href="https://github.com/gokceneraslan/awesome-deepbio">Awesome DeepBio</a></li>
                    <li><a href="http://colah.github.io/posts/2015-09-Visual-Information/">Visual Information Theory</a></li>
                </ul>
            </td>
        </tr>
        <tr>
            <td>Lecture 2</td>
            <td>Feb 7</td>
            <td>Neural networks and TensorFlow</td>
            <td>
                <ul>
                    <li>Read <a href="http://www.deeplearningbook.org/">Goodfellow</a> Chapter 6</li>
                    <li><a href="../assets/slides/FeedForwardBackprop.pdf">Feed Forward Backprop</a></li>
                </ul>
            </td>
            <td>
                <ul>
                    <li><a href="https://github.com/aymericdamien/TensorFlow-Examples">Damien</a></li>
                    <li><a href="http://nicklocascio.com/tensorflow-crash-course">Nick Locascio</a></li>
                    <li><a href="https://www.tensorflow.org/versions/r0.11/tutorials/index.html">TF site tutorials</a></li>
                </ul>
            </td>
        </tr>
        <tr>
            <td>Recitation 1</td>
            <td>Feb 7 / Feb 8</td>
            <td>Machine learning overview</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td>Lecture 3</td>
            <td>Feb 12</td>
            <td>Convolutional and recurrent neural networks</td>
            <td>
                <ul>
                    <li>Read <a href="http://www.deeplearningbook.org/">Goodfellow</a> Chapter 9 and 10</li>
                </ul>
            </td>
            <td></td>
        </tr>
        <tr>
            <td>Lecture 4</td>
            <td>Feb 14</td>
            <td>Introduction to protein-DNA interactions and ChIP-seq motif discovery</td>
            <td>TBD</td>
            <td></td>
        </tr>
        <tr class="warning">
            <td>Deadline</td>
            <td>Feb 14</td>
            <td>PS1 due</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td>Recitation 2</td>
            <td>Feb 14</td>
            <td>Neural networks and TensorFlow</td>
            <td></td>
            <td></td>
        </tr>
        <tr class="info">
            <td>No class</td>
            <td>Feb 19</td>
            <td>President’s Day</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td>Lecture 5</td>
            <td>Feb 21</td>
            <td>Models of protein-DNA interactions</td>
            <td>TBD</td>
            <td></td>
        </tr>
        <tr>
            <td>Recitation 3</td>
            <td>Feb 21</td>
            <td>Motifs, ChIP-seq, models</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td>Lecture 6</td>
            <td>Feb 26</td>
            <td>ML model interpretation  (gradient methods, SIS)</td>
            <td>TBD</td>
            <td></td>
        </tr>
        <tr class="warning">
            <td>Deadline</td>
            <td>Feb 26</td>
            <td>PS2 due</td>
            <td></td>
            <td></td>
        </tr>
        <tr> 
            <td>Lecture 7</td>
            <td>Feb 28</td>
            <td>The expressed genome and RNA splicing (RNA-seq)</td>
            <td>TBD</td>
            <td>
                <ul>
                    <li><a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0881-8">A survey of best practices for RNA-seq data analysis</a></li>
                    <li><a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0702-5">Comparative assessment of methods</a></li>
                </ul>
            </td>
        </tr>
        <tr>
            <td>Recitation 4</td>
            <td>Feb 28</td>
            <td>Model interpretation, projects</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td>Lecture 8</td>
            <td>Mar 05</td>
            <td>PCA, dimensionality reduction (t-SNE), autoencoders</td>
            <td>TBD</td>
            <td></td>
        </tr>
        <tr>
            <td>Lecture 9</td>
            <td>Mar 07</td>
            <td>scRNA seq and cell labeling, adversarial methods</td>
            <td>TBD</td>
            <td></td>
        </tr>
        <tr>
            <td>Recitation 5</td>
            <td>Mar 07</td>
            <td>Compressed state representations</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td>Lecture 10</td>
            <td>Mar 12</td>
            <td>Modeling gene expression and regulation</td>
            <td>TBD</td>
            <td></td>
        </tr>
        <tr>
            <td>Lecture 11</td>
            <td>Mar 14</td>
            <td>Model uncertainty, significance, multiple hypothesis testing</td>
            <td>TBD</td>
            <td></td>
        </tr>
        <tr>
            <td>Recitation 6</td>
            <td>Mar 14</td>
            <td>Model selection and L1/L2 regularization</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td>Lecture 12</td>
            <td>Mar 19</td>
            <td>Chromatin accessibility and marks</td>
            <td>TBD</td>
            <td></td>
        </tr>
        <tr class="warning">
            <td>Deadline</td>
            <td>Mar 19</td>
            <td>PS3 due</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td>Lecture 13</td>
            <td>Mar 21</td>
            <td>Predicting chromatin accessibility</td>
            <td>TBD</td>
            <td></td>
        </tr>
        <tr class="info">
            <td>No class</td>
            <td>Mar 26</td>
            <td>Spring break</td>
            <td></td>
            <td></td>
        </tr>
        <tr class="info">
            <td>No class</td>
            <td>Mar 30</td>
            <td>Spring break</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td>Lecture 14</td>
            <td>Apr 2</td>
            <td>Discovering and predicting 3D genome interactions</td>
            <td>TBD</td>
            <td></td>
        </tr>
        <tr>
            <td>Lecture 15</td>
            <td>Apr 4</td>
            <td>eQTL prediction and variant prioritization</td>
            <td>TBD</td>
            <td></td>
        </tr>
        <tr>
            <td>Recitation 7</td>
            <td>Apr 4</td>
            <td>Metrics for accessibility accuracy</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td>Lecture 16<br />by Adrian Dalca</td>
            <td>Apr 9</td>
            <td>Imaging and genotype to phenotype</td>
            <td>TBD</td>
            <td></td>
        </tr>
        <tr>
            <td>Lecture 17</td>
            <td>Apr 11</td>
            <td>Generative models: gradients, autoencoders, GANs</td>
            <td>TBD</td>
            <td></td>
        </tr>
        <tr>
            <td>Recitation 8</td>
            <td>Apr 11</td>
            <td>Lead SNPs to linked SNPs, haplotype structure</td>
            <td></td>
            <td></td>
        </tr>
        <tr class="warning">
            <td>Deadline</td>
            <td>Apr 11</td>
            <td>Project proposal due</td>
            <td></td>
            <td></td>
        </tr>
        <tr class="warning">
            <td>Deadline</td>
            <td>Apr 11</td>
            <td>PS4 due</td>
            <td></td>
            <td></td>
        </tr>
        <tr class="info">
            <td>No class</td>
            <td>Apr 16</td>
            <td>Patriots' Day</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td>Lecture 18<br />by Haoyang Zeng</td>
            <td>Apr 18</td>
            <td>Deep Learning for eQTL and variant prediction</td>
            <td>TBD</td>
            <td></td>
        </tr>
        <tr>
            <td>Lecture 19</td>
            <td>Apr 23</td>
            <td>Therapeutic design</td>
            <td>TBD</td>
            <td></td>
        </tr>
        <tr>
            <td>Lecture 20</td>
            <td>Apr 25</td>
            <td>Exam review session</td>
            <td>TBD</td>
            <td></td>
        </tr>
        <tr>
            <td>Recitation 9</td>
            <td>Apr 25</td>
            <td>Generative models</td>
            <td></td>
            <td></td>
        </tr>
        <tr class="warning">
            <td>Deadline</td>
            <td>Apr 25</td>
            <td>PS5 due</td>
            <td></td>
            <td></td>
        </tr>
        <tr class="danger">
            <td>Exam</td>
            <td>Apr 30</td>
            <td>Exam</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td></td>
            <td>May 2</td>
            <td>Project work</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td></td>
            <td>May 7</td>
            <td>Project work</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td></td>
            <td>May 9</td>
            <td>Project work</td>
            <td></td>
            <td></td>
        </tr>
        <tr class="warning">
            <td>Deadline</td>
            <td>May 9</td>
            <td>Project deck due</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td></td>
            <td>May 14</td>
            <td>Project presentations I</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td></td>
            <td>May 16</td>
            <td>Project presentations II</td>
            <td></td>
            <td></td>
        </tr>
    </table>

    <h2 class="module-header">Tutorials for TensorFlow, NumPy, Google Cloud, and Jupyter notebooks</h2>
    <p>We collected a series of pointers to tutorials on NumPy, TensorFlow, Google Cloud and Conda <a href="tutorials/main/">here</a>. We also provide a <a href="tutorials/quickstart/">Quickstart tutorial</a> to set up essential environment and tools for you to work on problem set 1.</p>
  
    <h2 class="module-header">Prerequisites</h2>
	<p>You should be comfortable with calculus, linear algebra, (Python) programming, probability, and introductory molecular biology. This will be a fast paced course, and it is targeted towards students that are both mathematically and computational capable. There are many other subjects at MIT that teach overviews of computational biology that are less demanding, we would be happy to recommend other options if you find this subject is not what you desire.</p>
    
    <h2 class="module-header">Class meeting times and places</h2>
    <ul>
        <li>Lecture: TR1-2.30 (32-155)</li>
        <li>Recitation: TBD</li>
    </ul>

    <h2 class="module-header">Contact</h2>
    <p>You should feel free to contact the lecturer and the TAs about any questions through <a href="mailto:6.874staff@mit.edu">6.874staff@mit.edu</a>. The best way to get detailed questions answered is to attend TA office hours and recitation or post them on <a target="_blank" href="https://piazza.com/class/jrawmzc9aph2m">Piazza</a>.</p>

    <h2 class="module-header">Office hours</h2>
    <div class="materials-item" style="border-top: 1px solid #CCC;">
        David Gifford (gifford@mit.edu): Office hours by appointment
    </div>
    <div class="materials-item">
        Siddhartha Jain (sjX@mit.edu) where X="1": TBD (Stata G5 lounge)
    </div>
    <div class="materials-item">
        Konstantin Krismer (krismer@mit.edu): TBD (Stata G5 lounge)
    </div>
    <div class="materials-item">
        Ge (Saber) Liu (geliu@mit.edu): TBD (Stata G5 lounge)
    </div>

    <h2 class="module-header">Grading</h2>
    <p>Grading will be based upon five programming-intensive problem sets (40%), a comprehensive exam (30%), and a final project (30%). Class participation both in lecture and recitation will be used to adjust grades. Attendance in lecture is important as the class moves quickly and you will need to be present. There will be an option for some students to give presentations on relevant research papers from the syllabus list. If you opt to present a paper in class it will be graded and substitute for a problem set of your choice. For students enrolled in one of the graduate versions of this class (6.874, 20.490, and HST.506) there will be an extra section on some problem sets. You can use three late days for problem set deadlines (or email from Student Support Services - S3).</p>

    <h2 class="module-header">Project</h2>
    <p>This subject has a substantial deep learning project component. Projects will be completed either as a team of two or individually. 6.874 students must complete an individual project unless permission from the teaching staff is given to work as a team. You are free to choose a problem in the life sciences and develop a deep learning solution using the subject’s cloud resources. Ideas will be provided along with example data, and you can consult the teaching staff to get advice for other ideas or to refine your own ideas.</p>

    <h2 class="module-header">Textbook</h2>
    <p>We will be using the book “Deep Learning” by Goodfellow, Bengio, and Courville. You can find the book online <a target="_blank" href="http://deeplearningbook.org">here</a> and <a target="_blank" href="https://github.com/janishar/mit-deep-learning-book-pdf">here</a>. You can purchase a hard copy at <a target="_blank" href="https://mitpress.mit.edu/books/deep-learning">MIT Press</a> or on Amazon.</p>
  </div>
</div>
